Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -727,7 +730,8 @@ internal async Task PerformActionTypeBehavior(TreeNode treeNode, string treeNode
                 }
 
                 // Break out early if we would hit timeout before next retry.
-                if (actionTimeout != -1 && stopwatch.ElapsedMilliseconds + waitTime.TotalMilliseconds >= actionTimeout)
+                // RetryPolicyType.FixedCount works independently of retry timeout
+                if (retryPolicyType != RetryPolicyType.FixedCount && actionTimeout != -1 && stopwatch.ElapsedMilliseconds + waitTime.TotalMilliseconds >= actionTimeout)
```

Comment:
Let's allow Timeout to be used with FixedCount. In general it may not be used, in which case the existing code should work with the default -1 infinite timeout. But if the user desires a timeout as well as a fixed count, they should be allowed specify that.

Reply:
I think we can remove the new code from here. FixedCount shouldn't require any special logic or comments. Timeout should work the same for any RetryPolicyType.

---
In reply to: [401966278](https://github.com/microsoft/Forge/pull/13#discussion_r401966278) [](ancestors = 401966278)

Reply:
Remove "retryPolicyType != RetryPolicyType.FixedCount" still in iteration 4. We want all RetryPolicyTypes to hit Timeout if it is set.

---
In reply to: [403294303](https://github.com/microsoft/Forge/pull/13#discussion_r403294303) [](ancestors = 403294303,401966278)